File: javascript/creational/factory.test.js
const {Car, Truck, HcmCarFactory, HnTruckFactory} = require("./factory");

test("Vehicle Factory test", () => {
    const hcmFactory = new HcmCarFactory();
    const hnFactory = new HnTruckFactory();
    hcmFactory.produce("Hcm");
    hnFactory.produce("Hn");
    const hcmCar = hcmFactory.getVehicle();
    const hnTruck = hnFactory.getVehicle();
    console.log(hcmCar.getName());
    console.log(hnTruck.getName());
    expect(hcmCar.getName()).toBe(`Car from Hcm produced at: ${new Date().getHours()}`);
    expect(hnTruck.getName()).toBe(`Truck from Hn produced at: ${new Date().getHours()}`);
  });
  

File: go/creational/factory_test.go
package creational

import "testing"

func TestFactory(t *testing.T) {
	f := &Factory{Name: "Factory"}
	car, err := f.GetVehicle("car")
	if err != nil {
		t.Error(err)
	}

	if car.GetName() != "Car4" {
		t.Errorf("Expected car.GetName() == Car4, but got %v", car.GetName())
	}

	truck, err := f.GetVehicle("truck")
	if err != nil {
		t.Error(err)
	}

	if truck.GetName() != "Truck" {
		t.Errorf("Expected truck.GetName() == Truck, but got %v", truck.GetName())
	}
}
